protocol/synthetix/subgraph/tests/handleDelegationUpdated.ts

```diff
@@ -61,5 +62,18 @@ export default function test(): void {
   assert.fieldEquals('Position', `1-1-${address}`, 'collateral_amount', '10000');
   assert.fieldEquals('Vault', `1-${address}`, 'updated_at', (now + 1000).toString());
   assert.fieldEquals('Vault', `1-${address}`, 'updated_at_block', now.toString());
-  assert.fieldEquals('Vault', `1-${address}`, 'collateral_amount', '-5354');
+  assert.fieldEquals('Vault', `1-${address}`, 'collateral_amount', '10000');
+
+  const newDelegatioNUpdatedEvent3 = createDelegationUpdateEvent(
+    BigInt.fromI32(2),
+    BigInt.fromI32(1),
+    Address.fromString(address),
+    BigInt.fromI32(5000),
+    BigInt.fromI32(10),
+    now + 1000,
+    now
+  );
+  handleDelegationUpdated(newDelegatioNUpdatedEvent3);
+
+  assert.fieldEquals('Vault', `1-${address}`, 'collateral_amount', '15000');
```

Comment:
```suggestion
  assert.fieldEquals('Vault', `1-${address}`, 'collateral_amount', '5000');
```
We have basically one staker that updated delegation from 10000 to 5000 (so delta is `-5000`, vault was 10000 before, so new vault amount should be 5000 now, not 15000

Reply:
Ideally this test now needs to have at least two different addresses with different positions so we can see that vault collateral amount is different from each individual position amount

Reply:
no, different account id

Reply:
so I think it should add it

Reply:
> Ideally this test now needs to have at least two different addresses with different positions so we can see that vault collateral amount is different from each individual position amount

yeah with account id set to 2 in event 3 we have that case

Reply:
Yeah ok. It is not clear what all the args are. Would be nice to have it written down like

```
var accountId1 = 2
createDelegationUpdateEvent(
    accountId1,
    smth,
    address1,
    ...
```

Another thing to do is to remove all the type conversions from test itself. They should be done in event creator. You can see all the example in existing event creators and tests. They do not have type casting in test code itself

Reply:
![image](https://github.com/Synthetixio/synthetix-v3/assets/28145325/48edee69-f168-44ed-b514-5798b7833574)

Reply:
look at spot/perps subgraph tests. This particular one was quite old and was not rewritten by me at the time of all the refactorings.

Reply:
Reading test now is hard, I could not figure out from test code that there were multiple positions. And all the new additions need to follow much more readable standard instead of adding a lot of a much less understandable legacy code.

Reply:
> look at spot/perps subgraph tests. This particular one was quite old and was not rewritten by me at the time of all the refactorings.

cant follow you? This is not about blaming someone. Refactoring the type conversions is not super urgent and important and can be tackled later on

Reply:
Not adding code that needs refactoring and adding code that does not need immediate refactoring is the best approach.

Reply:
"not super urgent" means it will stay like this forever